Sources/MongoSwift/Collection.swift

```diff
@@ -386,20 +400,43 @@ public struct IndexModel: BsonEncodable {
         return String(cString: mongoc_collection_keys_to_index_string(self.keys.data))
     }
 
-    public func encode(to encoder: BsonEncoder) throws {
-        // we need a flat document containing key, name, and options,
-        // so encode the options directly to this encoder first
-        try self.options?.encode(to: encoder)
-        try encoder.encode(keys, forKey: "key")
-        if self.options?.name == nil {
-            try encoder.encode(self.defaultName, forKey: "name")
-        }
+    // Encode own data as well as nested options data
+    private enum CodingKeys: String, CodingKey {
+        case key, background, expireAfter, name, sparse, storageEngine, unique,
+            version, defaultLanguage, languageOverride, textVersion, weights,
+            sphereVersion, bits, max, min, bucketSize, partialFilterExpression, collation
+    }
+
+    public func encode(to encoder: Encoder) throws {
```

Comment:
this is annoying, but we have to send the data with the options stored on the level as the index keys, so I couldn't figure out a nicer way to do this